File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/common/ContactCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.common;

import java.util.List;
import ma.zs.alc.bean.core.common.Contact;
import ma.zs.alc.dao.criteria.core.common.ContactCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ContactCollaboratorService extends  IService<Contact,ContactCriteria>  {

    List<Contact> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}

File: frontend-bqckup/src/app/shared/service/collaborator/common/ContactCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {ContactDto} from '../../../model/common/Contact.model';
import {ContactCriteria} from '../../../criteria/common/ContactCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class ContactCollaboratorService extends AbstractService<ContactDto, ContactCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/contact/';
    }

    public constrcutDto(): ContactDto {
        return new ContactDto();
    }

    public constrcutCriteria(): ContactCriteria {
        return new ContactCriteria();
    }
}
